acvm_backend_barretenberg/src/composer.rs

```diff
@@ -104,33 +104,32 @@ impl Composer for Barretenberg {
         unsafe {
             vk_size = barretenberg_sys::composer::init_verification_key(
                 pippenger_ptr,
-                &g2_data,
+                g2_data,
                 &proving_key,
                 vk_ptr,
             )
         }
 
-        std::mem::forget(g2_data);
         std::mem::forget(proving_key);
 
         let result;
         unsafe {
             result = Vec::from_raw_parts(vk_addr, vk_size, vk_size);
         }
-        result.to_vec()
+        result
     }
 
     fn create_proof_with_pk(
         &self,
+        reference_string: &CRS,
         constraint_system: &ConstraintSystem,
         witness: WitnessAssignments,
         proving_key: &[u8],
     ) -> Vec<u8> {
-        let circuit_size = self.get_circuit_size(constraint_system);
```

Comment:
I didn't like the idea of persisting it on the struct because then you have to do like `Option<CRS>` and check to see if you have it in each function. The idea behind `get_reference_string` is that it won't be called by nargo if Nargo already has a cached version that is large enough for the circuit, so we'd have to check/cache it in each function, which seems messy.